cobra/ui/model.go

```diff
@@ -0,0 +1,365 @@
+						var filesList []string
+						for _, file := range files {
+							fileInfo, err := os.Stat(file)
+							if err == nil && !fileInfo.IsDir() {
+								filesList = append(filesList, file)
+							}
+						}
+						if len(filesList) > 0 {
+							fileIndex := u.cursor
+							if fileIndex < len(filesList) {
+								selectedFile = filesList[fileIndex]
+							}
+						}
+						err = u.questions[u.index].Handler(selectedFile)
+					} else {
+						u.errorMsg = "Directory does not exist"
+					}
+				} else if len(u.questions[u.index].Options) > 0 {
+					err = u.questions[u.index].Handler(u.questions[u.index].Options[u.cursor])
+				} else {
+					err = u.questions[u.index].Handler(u.input)
+				}
+				if err != nil {
+					u.errorMsg = err.Error()
+					u.input = ""
+					u.cursor = 0
+					return u, nil
+				}
+				u.userAnswers = append(u.userAnswers, u.input)
+				u.input = ""
+				u.index++
+				u.cursor = 0
+				return u, nil
+			} else {
+				return nil, tea.Quit
+			}
+		case tea.KeyDown:
+			if u.questions[u.index].FilePath {
+				if _, err := os.Stat(u.input); err == nil {
+					files, _ := filepath.Glob(filepath.Join(u.input, "*"))
+					var filesList []string
+					for _, file := range files {
+						fileInfo, err := os.Stat(file)
+						if err == nil && !fileInfo.IsDir() {
+							filesList = append(filesList, file)
+						}
+					}
+					if len(filesList) > 0 {
+						if u.cursor < len(filesList)-1 {
+							u.cursor++
+						}
+					}
+				}
+			} else if len(u.questions[u.index].Options) > 0 {
+				if u.cursor < len(u.questions[u.index].Options)-1 {
+					u.cursor++
+				}
+			}
+
+		case tea.KeyUp:
+			if u.cursor > 0 {
+				u.cursor--
+			}
+		case tea.KeyRight:
+			if u.questions[u.index].FilePath {
+				if _, err := os.Stat(u.input); err == nil {
+					files, _ := filepath.Glob(filepath.Join(u.input, "*"))
+					page := u.cursor + pageSize
+					if page >= len(files) {
+						u.cursor = 0
+					} else {
+						u.cursor = page
+					}
+				}
+			} else if len(u.questions[u.index].Options) > 0 {
+				nextPage := (u.cursor/pageSize + 1) * pageSize
+				if nextPage < len(u.questions[u.index].Options) {
+					u.cursor = nextPage
+				} else {
+					u.cursor = len(u.questions[u.index].Options) - 1
+				}
+			}
+		case tea.KeyLeft:
+			if u.cursor >= pageSize {
+				u.cursor -= pageSize
+			} else {
+				u.cursor = 0
+			}
+
+		case tea.KeyTab:
+		case tea.KeySpace:
+		case tea.KeyBackspace:
+			if len(u.input) > 0 {
+				u.input = u.input[:len(u.input)-1]
+			}
+		default:
+			u.input += msg.String()
+		}
+	}
+
+	return u, nil
+}
+
+func (u *ui) View() string {
+	var dir string
+	if u.index >= len(u.questions) {
+		return u.LastMessage
+	}
+	view := u.questions[u.index].Text
+
+	if u.questions[u.index].FilePath {
+		dir += u.input
+		view += u.input + "\n"
+		if u.errorMsg != "" {
+			view += "Error: " + u.errorMsg + "\n"
+		}
+		fileInfo, err := os.Stat(u.input)
+		if err == nil && fileInfo.IsDir() {
```

Comment:
if you store in memory a complete list of file, print it completly
if you manage page, read only awaiting page into memory to prevent memory consumption